cmd/cost/get.go

```diff
@@ -0,0 +1,230 @@
+package cost
+
+import (
+	"fmt"
+	awsprovider "github.com/openshift/osd-utils-cli/pkg/provider/aws"
+	"github.com/spf13/cobra"
+	"log"
+	"strconv"
+	"time"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/costexplorer"
+	"github.com/aws/aws-sdk-go/service/organizations"
+)
+
+// getCmd represents the get command
+func newCmdGet(streams genericclioptions.IOStreams) *cobra.Command {
+	ops := newGetOptions(streams)
+	getCmd := &cobra.Command{
+		Use:   "get",
+		Short: "Get total cost of a given OU. If no OU given, then gets total cost of v4 OU.",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			awsClient, err := opsCost.initAWSClients()
+			cmdutil.CheckErr(err)
+
+			//Get Organizational Unit
+			OU := organizations.OrganizationalUnit{Id: aws.String(ops.ou)}
+			//Store cost
+			var cost float64 = 0
+
+			if ops.recursive { //Get cost of given OU by aggregating costs of all (including immediate) accounts under OU
+				getOUCostRecursive(&OU, awsClient, &ops.time, &cost)
+				fmt.Printf("Cost of %s recursively is: %f\n", ops.ou, cost)
+			} else { //Get cost of given OU by aggregating costs of only immediate accounts under given OU
+				getOUCost(&OU, awsClient, &ops.time, &cost)
+				fmt.Printf("Cost of %s is: %f\n", ops.ou, cost)
+			}
+		},
+	}
+	getCmd.Flags().StringVar(&ops.ou, "ou", "ou-0wd6-aff5ji37", "get OU ID (default is v4)") //Default OU is v4
+	getCmd.Flags().BoolVarP(&ops.recursive, "recursive", "r", false, "recurse through OUs")
+	getCmd.Flags().StringVarP(&ops.time, "time", "t", "", "set time")
+
+	return getCmd
+}
+
+//Store flag options for get command
+type getOptions struct {
+	ou        string
+	recursive bool
+	time      string
+
+	genericclioptions.IOStreams
+}
+
+func newGetOptions(streams genericclioptions.IOStreams) *getOptions {
+	return &getOptions{
+		IOStreams: streams,
+	}
+}
+
+//Get account IDs of immediate accounts under given OU
+func getAccounts(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client) []*string {
+	var accountSlice []*string
+	var nextToken *string
+
+	//Populate accountSlice with accounts by looping until accounts.NextToken is null
+	for {
+		accounts, err := awsClient.ListAccountsForParent(&organizations.ListAccountsForParentInput{
+			ParentId:  OU.Id,
+			NextToken: nextToken,
+		})
+		if err != nil {
+			log.Fatalln("Unable to retrieve accounts under OU:", err)
```

Comment:
I think this needs a return. Otherwise line 81 may try to iterate over a nil pointer.